File: 04) PHP Web Development Basics/10) OOP Basics - Exercises/8. Pokemon Trainer/Pokemon.php
<?php

namespace Van\Classes;

class Pokemon
{
    private $name;

    private $element;

    private $health;

    public function __construct(string $name, string $element, int $health)
    {
        $this->name = $name;
        $this->element = $element;
        $this->health = $health;
    }

    public function getElement(): string
    {
        return $this->element;
    }

    public function loseHealth(int $healthTaken)
    {
        $this->health -= $healthTaken;
    }

    public function getHealth(): int
    {
        return $this->health;
    }
}

File: Software_University/09_Java_Advanced_January_2019/14_Defining_Classes_Exercises/06_Pokemon_Trainer/Pokemon/Pokemon.java
package Pokemon;

public class Pokemon {

    private String name;

    private String element;

    private int health;

    public Pokemon(String name, String element, int health) {
        setName(name);
        setElement(element);
        setHealth(health);
    }

    public void takeDamage(int damageToTake) {
        setHealth(health - damageToTake);
    }

    public boolean isAlive() {
        return health > 0;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getElement() {
        return element;
    }

    public void setElement(String element) {
        this.element = element;
    }

    public int getHealth() {
        return health;
    }

    public void setHealth(int health) {
        this.health = health;
    }


}
